gno.land/pkg/gnoweb/weburl/url.go

```diff
@@ -226,11 +218,12 @@ func ParseGnoURL(u *url.URL) (*GnoURL, error) {
 	}, nil
 }
 
-// NoEscapeQuery generates a URL-encoded query string from the given url.Values,
-// without escaping the keys and values. The query parameters are sorted by key.
-func NoEscapeQuery(v url.Values) string {
-	// Encode encodes the values into “URL encoded” form
-	// ("bar=baz&foo=quux") sorted by key.
+// BuildQueryString generates a URL-encoded query string from the given url.Values.
+// This function is a modified version of Go's `url.Values.Encode()`: https://pkg.go.dev/net/url#Values.Encode
+// - If `escapeKey` is `true`, keys and values are escaped using `QueryEscape()` (same as `Values.Encode()`).
+// - If `escapeKey` is `false`, keys and values are **not escaped** (similar to `NoEscapeQuery()`).
+// - Empty string values do not get an `=` sign (`?key` instead of `?key=`).
+func BuildQueryString(v url.Values, escape bool) string {
```

Comment:
I think `Encode` is more appropriate than `Build` as the function retains most of the logic of the original Encode method.
I've also added some suggestions to the whole text comment.

```suggestion
// EncodeQuery generates a URL-encoded query string from the given url.Values.  
// This function is a modified version of Go's `url.Values.Encode()`: https://pkg.go.dev/net/url#Values.Encode  
// It takes an additional `escape` boolean argument that disables escaping on keys and values.  
// Additionally, if an empty string value is passed, it omits the `=` sign, resulting in `?key` instead of `?key=` to enhance URL readability.
func EncodeQuery(v url.Values, escape bool) string {
```